adapter/integration-test/src/adapter-datasource.test.js

```diff
@@ -1,77 +1,125 @@
 
 const {
-  ADAPTER_URL
+  ADAPTER_URL,
+  AMQP_URL
 } = require('./env')
 const { waitForServicesToBeReady } = require('./waitForServices')
+const {
+  connectAmqp,
+  consumeAmqpMsg
+} = require('./testHelper')
+
+const AMQP_EXCHANGE = 'ods_global'
+const AMQP_QUEUE = 'adapter-it-ds'
+const CONFIG_TOPIC = 'datasource.config.*'
+const CONFIG_CREATED_TOPIC = 'datasource.config.created'
+const CONFIG_DELETED_TOPIC = 'datasource.config.deleted'
+const CONFIG_UPDATED_TOPIC = 'datasource.config.updated'
 
 const TIMEOUT = 10000
+const publishedEvents = new Map() // routing key -> received msgs
+let amqpConnection
 
 describe('Datasource Configuration', () => {
   beforeAll(async () => {
     await waitForServicesToBeReady()
+
+    amqpConnection = await connectAmqp(AMQP_URL)
+
+    await consumeAmqpMsg(amqpConnection, AMQP_EXCHANGE, CONFIG_TOPIC, AMQP_QUEUE, publishedEvents)
   }, 60000)
 
+  afterAll(async () => {
+    await request(ADAPTER_URL)
+      .delete('/configs')
+      .send()
+    if (amqpConnection) {
+      await amqpConnection.close()
+    }
+  }, TIMEOUT)
+
   test('Should respond with stored datasource configs', async () => {
     const response = await request(ADAPTER_URL).get('/datasources')
     expect(response.status).toEqual(200)
     expect(response.type).toEqual('application/json')
 
-    expect(response.body).toEqual([])
+    expect(response.body).toBeInstanceOf(Array)
   }, TIMEOUT)
 
+  test('Should reject datasources with specified id [POST /datasources]', async () => {
+    const configWithId = Object.assign({}, datasourceConfig)
+    configWithId.id = 1
+    const response = await request(ADAPTER_URL)
+      .post('/datasources')
+      .send(configWithId)
+    expect(response.status).toEqual(400)
+  })
+
   test('Should create datasources [POST /datasources]', async () => {
     const response = await request(ADAPTER_URL)
       .post('/datasources')
       .send(datasourceConfig)
+    const datasource = response.body
 
     expect(response.status).toEqual(201)
     expect(response.header.location).toContain(response.body.id)
     expect(response.body.adapter).toEqual(datasourceConfig.adapter)
     expect(response.body.trigger).toEqual(datasourceConfig.trigger)
-    expect(response.body.id).toBeDefined()
-    expect(response.body.id).not.toEqual(datasourceConfig.id) // id not under control of client
+    expect(response.body.id).toBeGreaterThan(0)
 
-    const delResponse = await request(ADAPTER_URL)
-      .delete('/datasources/' + response.body.id)
-      .send()
-
-    expect(delResponse.status).toEqual(204)
+    expect(publishedEvents.get(CONFIG_CREATED_TOPIC)).toContainEqual({
+      datasource
+    })
   }, TIMEOUT)
 
   test('Should update existing datasource [PUT /datasources/{id}]', async () => {
     const postResponse = await request(ADAPTER_URL)
       .post('/datasources')
       .send(datasourceConfig)
-
-    const datasourceId = postResponse.body.id
+    const datasource = postResponse.body
```

Comment:
better name createdDatasource. It's hard to tell what the difference between datasource and datasourceConfig.